apps/POLite/util/polite.mk

```diff
@@ -46,13 +52,39 @@ $(INC)/config.h: $(TINSEL_ROOT)/config.py
 $(HL)/hostlink.a:
 	make -C $(HL) hostlink.a
 
-$(BUILD)/run: $(RUN_CPP) $(RUN_H) $(HL)/hostlink.a
-	g++ -std=c++11 -O2 -I $(INC) -I $(HL) -o $(BUILD)/run $(RUN_CPP) $(HL)/hostlink.a \
-	  -lmetis -fno-exceptions -fopenmp
+RUN_CPP_OBJ := $(patsubst %.cpp, build/%.run.o, $(RUN_CPP))
+SIM_CPP_OBJ := $(patsubst %.cpp, build/%.sim.o, $(RUN_CPP))
+SIM_RELEASE_CPP_OBJ := $(patsubst %.cpp, build/%.sim-release.o, $(RUN_CPP))
+
+$(BUILD)/%.run.o : %.cpp
+	mkdir -p $(BUILD)
+	g++ -c -std=c++11 -I $(INC) -I $(HL) -o $(BUILD)/$*.run.o $*.cpp \
+	   -std=c++17 -march=native -g -O3 -DNDEBUG=1 -fno-exceptions -fopenmp \
+	   -fno-omit-frame-pointer
+
+$(BUILD)/run: $(RUN_CPP_OBJ) $(RUN_H) $(HL)/hostlink.a
+	g++ -std=c++11 -o $(BUILD)/run $(RUN_CPP_OBJ) $(HL)/hostlink.a \
+	   -std=c++17 -march=native -g -O3 -DNDEBUG=1 -fno-exceptions -fopenmp \
+	   -fno-omit-frame-pointer -ltbb -lmetis
+
+
+POLITE_SW_SIM_DIR = $(TINSEL_ROOT)/apps/POLite/util/POLiteSWSim
+
+$(BUILD)/%.sim.o : %.cpp
+	mkdir -p $(BUILD)
+	g++ -std=c++17 -g -c -O0 -I $(POLITE_SW_SIM_DIR)/include -I $(INC) -I $(HL) -o $(BUILD)/$*.sim.o $*.cpp \
+		-fopenmp
+
+$(BUILD)/sim: $(SIM_CPP_OBJ) $(RUN_H)
+	g++ -std=c++17 -g $(SIM_CPP_OBJ) -o $(BUILD)/sim -g -lmetis -ltbb -fopenmp
+
+$(BUILD)/%.sim-release.o : %.cpp
+	mkdir -p $(BUILD)
+	g++ -std=c++17 -g -c -O3 -DNDEBUG=1 -I $(POLITE_SW_SIM_DIR)/include -I $(INC) -I $(HL) -o $(BUILD)/$*.sim.o $*.cpp \
+		-fopenmp
 
-$(BUILD)/sim: $(RUN_CPP) $(RUN_H) $(HL)/sim/hostlink.a
-	g++ -O2 -I $(INC) -I $(HL) -o $(BUILD)/sim $(RUN_CPP) $(HL)/sim/hostlink.a \
-    -lmetis
```

Comment:
HI @m8pple, this target was/is useful for running POLite apps on top of the hardware (Bluesim) simulator.  Would be nice if this still works.  I don't mind this target is renamed (as it clashes with the new `sim` target for software simulation), but we probably don't want to remove it.